packages/docs/site/docs/developers/05-local-development/04-wp-playground-cli.md

```diff
@@ -321,3 +437,325 @@ cliServer = await runCLI({
+```TypeScript
+import { runCLI } from "@wp-playground/cli";
+
+const cliServer = await runCLI({
+  command: 'server',
+  'experimental-blueprints-v2-runner': true,
+  mode: 'create-new-site',
+  'mount-before-install': [
+    {
+      hostPath: './my-new-site',
+      vfsPath: '/wordpress'
+    }
+  ]
+});
+```
+
+**Applying to an existing site:**
+
+```TypeScript
+const cliServer = await runCLI({
+  command: 'server',
+  'experimental-blueprints-v2-runner': true,
+  mode: 'apply-to-existing-site',
+  'mount-before-install': [
+    {
+      hostPath: './existing-wordpress',
+      vfsPath: '/wordpress'
+    }
+  ],
+  blueprint: {
+    steps: [
+      {
+        step: 'setSiteOptions',
+        options: {
+          blogname: 'Updated Site Name'
+        }
+      }
+    ]
+  }
+});
+```
+
+## Automated testing
+
+### Integration testing with Vitest
+
+The programmatic API is excellent for automated testing. Here's a complete example using Vitest:
+
+```TypeScript
+import { describe, test, expect, afterEach } from 'vitest';
+import { runCLI, RunCLIServer } from "@wp-playground/cli";
+
+describe('My Plugin Tests', () => {
+  let cliServer: RunCLIServer;
+
+  afterEach(async () => {
+    if (cliServer) {
+      await cliServer[Symbol.asyncDispose]();
+    }
+  });
+
+  test('plugin activates successfully', async () => {
+    cliServer = await runCLI({
+      command: 'server',
+      mount: [
+        {
+          hostPath: './my-plugin',
+          vfsPath: '/wordpress/wp-content/plugins/my-plugin'
+        }
+      ],
+      blueprint: {
+        steps: [
+          {
+            step: 'activatePlugin',
+            pluginPath: '/wordpress/wp-content/plugins/my-plugin/plugin.php'
+          }
+        ]
+      }
+    });
+
+    const homeUrl = new URL('/', cliServer.serverUrl);
+    const response = await fetch(homeUrl);
+
+    expect(response.status).toBe(200);
+    const html = await response.text();
+    expect(html).toContain('My Plugin');
+  });
+
+  test('plugin settings page loads', async () => {
+    cliServer = await runCLI({
+      command: 'server',
+      login: true, // Auto-login as admin
+      mount: [
+        {
+          hostPath: './my-plugin',
+          vfsPath: '/wordpress/wp-content/plugins/my-plugin'
+        }
+      ],
+      blueprint: {
+        steps: [
+          {
+            step: 'activatePlugin',
+            pluginPath: '/wordpress/wp-content/plugins/my-plugin/plugin.php'
+          }
+        ]
+      }
+    });
+
+    const settingsUrl = new URL(
+      '/wp-admin/options-general.php?page=my-plugin',
+      cliServer.serverUrl
+    );
+    const response = await fetch(settingsUrl);
+
+    expect(response.status).toBe(200);
+  });
+});
+```
+
+### Testing theme customizations
+
+```TypeScript
+test('theme displays custom header', async () => {
+  cliServer = await runCLI({
+    command: 'server',
+    mount: [
+      {
+        hostPath: './my-theme',
+        vfsPath: '/wordpress/wp-content/themes/my-theme'
```

Comment:
This isn't much different from one of the first examples. Also, I can't just copy&paste this snippet since I don't have `./my-theme` directory locally. Could we offer a few examples that can be just copy&pasted?